packages/libs/error-stack/src/report.rs

```diff
@@ -415,26 +443,39 @@ impl<T: Context> Report<T> {
 }
 
 impl<Context> fmt::Display for Report<Context> {
-    fn fmt(&self, fmt: &mut Formatter<'_>) -> fmt::Result {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
         #[cfg(feature = "hooks")]
-        if let Some(debug_hook) = Report::display_hook() {
-            return debug_hook(self.generalized(), fmt);
+        if let Some(display_hook) = Report::display_hook() {
+            return display_hook(self.generalized(), fmt);
         }
 
-        let mut chain = self.frames();
-        let error = chain.next().expect("No error occurred");
-        fmt::Display::fmt(&error, fmt)?;
-        if let Some(cause) = chain.next() {
-            if fmt.alternate() {
-                write!(fmt, ": {cause}")?;
+        for (index, frame) in self
+            .frames()
+            .filter_map(|frame| match frame.kind() {
+                FrameKind::Context(context) => Some(context.to_string()),
+                FrameKind::Attachment(Attachment::Printable(attachment)) => {
+                    Some(attachment.to_string())
+                }
+                FrameKind::Attachment(Attachment::Generic(_)) => None,
```

Comment:
I think we might want to say 
> And X non-displayable attachments

At the end (not printing a message for every single one)